File: Modulo02/ContaCorrente-CSharp/ContaChequeEspecial.cs
using System;

public class ContaChequeEspecial : ContaCorrente {

    private double credito;

    public ContaChequeEspecial(int numero, double saldo, double credito) : base (numero, saldo) {
        this.credito = credito;
    } 
    
    public override void imprime() {
        base.imprime();
        Console.WriteLine("Credito: {0:0.00}", this.credito);
    }
}

File: Modulo02/ContaCorrente-Cpp/ContaChequeEspecial.cpp
#include <iostream>
#include "ContaChequeEspecial.h"

using namespace std;

ContaChequeEspecial::ContaChequeEspecial(int numero, double saldo, double credito) : ContaCorrente(numero, saldo) {
    cout << "Construindo Conta Especial " << numero << endl;
    this->credito = credito;
}

ContaChequeEspecial::~ContaChequeEspecial() {
    cout << "Destruindo Conta Especial " << this->getNumero() << endl;
}

double ContaChequeEspecial::getCredito() const {
    return this->credito;
}

void ContaChequeEspecial::imprime() const {
    ContaCorrente::imprime();
    cout << "Credito: " << credito << endl;
}
